tests/test_gemm.py

```diff
@@ -0,0 +1,160 @@
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+import unittest
+import numpy as np
+import modmesh as mm
+
+
+class GemmTestBase(mm.testing.TestBase):
+    """Base class for matrix multiplication (GEMM) tests"""
+
+    def test_square_matrix_multiplication(self):
+        """Test basic square matrix multiplication"""
+        # Create 2x2 matrices
+        a_data = np.array([[1.0, 2.0], [3.0, 4.0]], dtype=self.dtype)
+        b_data = np.array([[5.0, 6.0], [7.0, 8.0]], dtype=self.dtype)
+
+        a = self.SimpleArray(array=a_data)
+        b = self.SimpleArray(array=b_data)
+
+        # Expected result: [[19, 22], [43, 50]]
+        expected = np.array([[19.0, 22.0], [43.0, 50.0]], dtype=self.dtype)
+
+        # Test matrix multiplication
+        result = a.matmul(b)
+
+        self.assertEqual(list(result.shape), [2, 2])
+        np.testing.assert_array_almost_equal(result.ndarray, expected)
+
+    def test_rectangular_matrix_multiplication(self):
+        """Test rectangular matrix multiplication"""
+        # Create 2x3 and 3x2 matrices
+        a_data = np.array([[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]],
+                          dtype=self.dtype)
+        b_data = np.array([[7.0, 8.0], [9.0, 10.0], [11.0, 12.0]],
+                          dtype=self.dtype)
+
+        a = self.SimpleArray(array=a_data)
+        b = self.SimpleArray(array=b_data)
+
+        # Expected result: [[58, 64], [139, 154]]
+        expected = np.array([[58.0, 64.0], [139.0, 154.0]],
+                            dtype=self.dtype)
+
+        result = a.matmul(b)
+
+        self.assertEqual(list(result.shape), [2, 2])
+        np.testing.assert_array_almost_equal(result.ndarray, expected)
+
+    def test_identity_matrix(self):
+        """Test multiplication with identity matrix"""
+        # 3x3 matrix
+        a_data = np.array([[1.0, 2.0, 3.0], [4.0, 5.0, 6.0],
+                           [7.0, 8.0, 9.0]], dtype=self.dtype)
+        identity_data = np.eye(3, dtype=self.dtype)
+
+        a = self.SimpleArray(array=a_data)
+        identity = self.SimpleArray(array=identity_data)
+
+        result = a.matmul(identity)
+
+        self.assertEqual(list(result.shape), [3, 3])
+        np.testing.assert_array_almost_equal(result.ndarray, a_data)
+
+    def test_zero_matrix(self):
+        """Test multiplication with zero matrix"""
+        a_data = np.array([[1.0, 2.0], [3.0, 4.0]], dtype=self.dtype)
+        zero_data = np.zeros((2, 2), dtype=self.dtype)
+
+        a = self.SimpleArray(array=a_data)
+        zero = self.SimpleArray(array=zero_data)
+
+        result = a.matmul(zero)
+
+        self.assertEqual(list(result.shape), [2, 2])
+        np.testing.assert_array_almost_equal(result.ndarray, zero_data)
+
+    def test_dimension_mismatch_error(self):
```

Comment:
Good to test the shape.  After adding shape information in the exception message, the testing code should also test the messages.